LocationUpdatesBackgroundKotlin/app/src/main/java/com/google/android/gms/location/sample/locationupdatesbackgroundkotlin/data/LocationRepository.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2020 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.gms.location.sample.locationupdatesbackgroundkotlin.data
+
+import android.content.Context
+
+import androidx.lifecycle.LiveData
+
+import java.util.UUID
+import java.util.concurrent.Executors
+
+import com.google.android.gms.location.sample.locationupdatesbackgroundkotlin.data.db.MyLocationDatabase
+import com.google.android.gms.location.sample.locationupdatesbackgroundkotlin.data.db.MyLocationEntity
+
+private const val TAG = "LocationRepository"
+
+/**
+ * Access point for database (MyLocation data) and location APIs (start/stop location tracking and
+ * checking tracking status).
+ */
+class LocationRepository private constructor(private val myLocationDatabase: MyLocationDatabase,
+                                             private val myLocationManager: MyLocationManager
+                                         ) {
+
+    // Database related fields/methods:
+    private val locationDao = myLocationDatabase.locationDao()
+
+    private val executor = Executors.newSingleThreadExecutor()
```

Comment:
because you're creating the executor here, it means that you're not able to unit test this class easily.
I suggest passing the executor as a constructor parameter as well. In tests, this can be replaced with an instant executor.

Reply:
ah, good call, moved.